### PR TITLE
#164797160 Users can see the time it takes to read an Article

### PR DESCRIPTION
**Pull request by emmsdan**
#### What does this PR do?
- wrote the test to check if users can read an article.
- users can see the time it takes to read an article

#### Description of Task to be completed?
none

#### Any background context you want to provide?
Initially, when reading an article there was no actual way to ascertain the possible amount of time for a reader to go through an article.

#### What are the relevant pivotal tracker stories?(if applicable)
#164797160 

#### Screenshots

Screenshot 2019-04-11 at 2 09 18 PM

### CODE REVIEW

File: src/utils/readtime.utils.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  return Math.floor(words.split(/[^\s]+/).length - 1);
+}
+
+/**
+ * @description calculate the time to view images in an article
+ * @param {array} images
+ * @returns {number} time in seconds for all images
+ */
+export function calcImageTime(images) {
+  let initSeconds = 11;
```
Comment (musonant) [inline src/utils/readtime.utils.js]: The variable `initSeconds` can be more descriptive, I think. From what I see here, it is supposed to be the time it takes to view an image.

File: src/utils/readtime.utils.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @param {object} content
+ * @returns {string} minutes to read article
+ */
+export function calculateTimeToReadArticle(content) {
+  const imageTime = calcImageTime(content.images);
+  const videoTime = calcVideoTime(content.videos);
+  const numberOfWords = wordCount(content.words);
+  const textTime = calcTextTime(numberOfWords);
+  const seconds = imageTime + videoTime + textTime;
+  const lex = moment.duration(seconds, 'seconds').format('hh:mm:ss');
```
Comment (eben-k) [inline src/utils/readtime.utils.js]: What's 'lex' please?

File: src/utils/readtime.utils.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  return totalSeconds;
+}
+
+/**
+ * @description calculate the total average time to read videos
+ * @param {array} videos list the various video durations ['1023', '23023']
+ * @returns {number} minutes to view all videos
+ */
+export function calcVideoTime(videos) {
+  let totalDuration = 0;
```
Comment (musonant) [inline src/utils/readtime.utils.js:39]: `videos` seems to be an array of the duration of each video. Using the variable name 'video' would become confusing. What do you think?

**Comment by oldmonad** [comment]
In the screenshot what is the unit of 1?

**Comment by emmsdan** [comment]
> In the screenshot what is the unit of 1?

It's in minutes. I did not include minutes (as string) on with eg. *1 min*, because in cases where the post contains multiple images lets say 100 images (image gallery), the minutes could span up to *80mins* which when converted can be around *1hour 20mins*. 
so my passing in only the digits in minutes gives the frontend engineers the liberty to handle the minute's returned as desired.

File: src/utils/readtime.utils.js (hunk 3 above)
Comment (emmsdan) [reply src/utils/readtime.utils.js:39]: Yeah, that's right. I would look into it.

File: src/utils/readtime.utils.js (hunk 2 above)
Comment (emmsdan) [reply src/utils/readtime.utils.js]: that's a typo error. I would remove it. Thanks for that

File: src/utils/readtime.utils.js (hunk 1 above)
Comment (emmsdan) [reply src/utils/readtime.utils.js]: thank, am on it.

File: src/utils/readtime.utils.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  return totalDuration;
+}
+
+/**
+ * @description calculate the total average time to read only text based article
+ * @param {number} seconds
+ * @returns {number} minutes to read article
+ */
+export function calcTextTime(seconds) {
```
Comment (Japhethca) [inline src/utils/readtime.utils.js]: I think `seconds` here should be `numberOfWords`.

Comment (emmsdan) [reply src/utils/readtime.utils.js]: thanks, I will fix that.

Comment (emmsdan) [reply src/utils/readtime.utils.js]: I have fixed it, change `seconds` variable to  `numberOfOwords`

**Review by Japhethca — APPROVED** [review]

**Pull request merged**